lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -512,6 +545,14 @@ class TraceProcessor {
     // Compute the key frame timings for the main frame.
     const frameTimings = this.computeKeyTimingsForFrame(frameEvents, {timeOriginEvt});
 
+    // Compute LCP for all frames.
+    const lcpAllFramesEvt = this.computeValidLCP({
+      candidateEventName: 'NavStartToLargestContentfulPaint::Candidate::AllFrames::UKM',
+      invalidateEventName: 'NavStartToLargestContentfulPaint::Invalidate::AllFrames::UKM',
+      events: keyEvents,
```

Comment:
I think there's an edge case this may not account for.

The situation would be theverge loading in one tab while nytimes is loading in another tab.  `keyEvents` is for all processes and so I think the resulting LCP-AF would merge together both tabs.

I'm not sure this is the case, but it looks like it...

the [trace event lives in `browser/`](https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/page_load_metrics/observers/core/ukm_page_load_metrics_observer.cc;l=1092-1104;drc=2b934611e4e480b7121488288f97ef27e0950e2a) so my guess is it's emitted on the browser process. it does contain a mainframeid however. and we have `findMainFrameIds`, so I think we can filter the trace events to be just the mainframeid ones we care about.

Reply:
> The situation would be theverge loading in one tab while nytimes is loading in another tab. keyEvents is for all processes and so I think the resulting LCP-AF would merge together both tabs.

I jumped to this as well, but I tried on half a dozen pages unsuccessfully 😅 Is the "looks like it" from looking at the code in Chromium or a case that happened?

Reply:
What would a trace like this look like?

Using the performance panel, I am able to produce a trace with LCP-AF events with two different values for `main_frame_tree_node_id`.

Reply:
hmm i just got one.

trace: https://gist.github.com/paulirish/ebfeeac8d2a243042041ab4e1bc87c7f

![image](https://user-images.githubusercontent.com/39191/100676758-5bf8ce80-331e-11eb-9805-8e83f69ff34d.png)


repro:

* `chrome-debug`
* open devtools
* open another chrome window, and make sure both windows are visible (not occluding the other)
* put example.com in window 1's omnibox and paulirish.com in window 2's.
* start perf panel recording
* hit enter on both omniboxes
* wait a bit and stop recording

Reply:
> and make sure both windows are visible (not occluding the other)

🤦 this is almost definitely what I forgot

Reply:
@paulirish `findMainFrameIds` doesn't return a node id, is there another way to get the main frame node id?

Reply:
> @paulirish `findMainFrameIds` doesn't return a node id, is there another way to get the main frame node id?

good question.... doing this would certainly be easier if `main_frame_tree_node_id` was instead `main_frame_tree_frame_id` but alas.  Hmmmm.. we'll have to do some hunting. 

@npm1 do you currently make use of `main_frame_tree_frame_id` to filter things? as of now, we don't have any mapping between frames/frametrees to node ids, so before we invent one, I'm curious if something already exists.